Framework/MDAlgorithms/src/IntegrateQLabEvents.cpp

```diff
@@ -301,6 +316,27 @@ PeakShapeEllipsoid_const_sptr IntegrateQLabEvents::ellipseIntegrateEvents(
   inti = peak_w_back.first - ratio * backgrd.first;
   sigi = sqrt(peak_w_back.second + ratio * ratio * backgrd.second);
 
+  if (inti < 0) {
+    std::ostringstream msg;
+    msg << "Negative intensity found: " << inti << "\n"
```

Comment:
```suggestion
    g_log.notice() << "Negative intensity found: " << inti << "\n"
```

Reply:
The logger is happy to split strings every time you call `<<` on it. That often interleaves log messages, making them hard to read. By only giving it whole log messages, this isn't an option for the logger.